src/org/openbravo/financial/ResetAccounting.java

```diff
@@ -384,14 +389,21 @@ private static HashMap<String, Integer> delete(final List<String> transactionIds
         OBDal.getInstance().getSession().clear();
       }
       return result;
-    } catch (final Exception e) {
+    } catch (Exception e) {
       OBDal.getInstance().rollbackAndClose();
-      throw new OBException("Error Deleting Accounting", e);
+      String errorMessage = getErrorMessage(e);
+      throw new OBException(String.format(OBMessageUtils.messageBD("Error_Deleting_Accounting"), errorMessage));
     } finally {
       OBContext.restorePreviousMode();
     }
   }
 
+  private static String getErrorMessage(Exception e) {
```

Comment:
It wouldn't be better to catch PersistenceException, and handle it accordingly.
And then catch the other exceptions, and handle them separatedly?

Reply:
Originally, I handled PersistenceException and Exception in separate catch blocks. However, this resulted in code duplication. To avoid this redundancy, I chose to use a single catch block and differentiate the exceptions with an if statement.